torchmultimodal/modules/layers/transformer.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+from collections import namedtuple
+from functools import partial
+from typing import Any, Callable, Optional
+
+import torch
+from torch import nn, Tensor
+from torchmultimodal.modules.layers.normalizations import Fp32LayerNorm
+
+TransformerOutput = namedtuple(
+    "TransformerOutput",
+    ["last_hidden_state", "pooler_output", "hidden_states", "attentions"],
+    defaults=(None, None, None, None),
+)
+
+
+class TransformerSelfAttention(nn.Module):
+    def __init__(
+        self,
+        hidden_size: int = 768,
+        num_attention_heads: int = 12,
+        attention_probs_dropout_prob: float = 0.0,
```

Comment:
Probably also take `**kwargs` at this point since you are refactoring.

Reply:
do we need `**kwargs` here?